src/any_agent/tools/mcp/mcp_server.py

```diff
@@ -33,18 +52,3 @@ def _check_dependencies(self) -> None:
 
         msg = f"You need to `pip install '{self.libraries}'` to use MCP."
         raise ImportError(msg)
-
-    def _filter_tools(self, tools: Sequence[Any]) -> Sequence[Any]:
-        # Only add the tools listed in mcp_tool['tools'] if specified
-        requested_tools = list(self.mcp_tool.tools or [])
-        if not requested_tools:
-            return tools
-        tools = [tool for tool in tools if tool.name in requested_tools]
```

Comment:
Link to #298  We missed this because we didn't have unit tests 🤦  but this logic was the thing doing the filtering of tools, which was accidentally removed in this PR